12 спринт финал

pavelrk97 opened the pull request:
Перевод работы с внутреннего хранения на работу с БД

Commit:
12 спринт финал
Перевод работы с внутреннего хранения на работу с БД

Commit:
12 спринт финал
Перевод работы с внутреннего хранения на работу с БД

Commit:
12 спринт финал
Смена типа ошибок у жанра и мпа на NotFoundException

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,32 @@
+
+    private final DbGenreServiceImpl service;
+
+    public GenreController(DbGenreServiceImpl service) {
```
Вместо шаблоного кода можно использовать ломбок, аналогично для остальных классов

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -29,6 +30,11 @@ public Collection<User> findAll() {
 
+    @GetMapping("/{id}")
+    public User findById(@PathVariable Long id) {
+        return service.findById(id).isPresent() ? service.findById(id).get() : null;
```
Лучше использовать Optional вместо null

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/mappers/FilmRowMappers.java:
```diff
@@ -0,0 +1,46 @@
+    private final FilmGenreStorage filmGenreStorage;
+    private final FilmLikeStorage filmLikeStorage;
+
+    public Film mapRowToFilm(ResultSet resultSet, int rowNum) throws SQLException {
```
Все-таки это не только маппер. Цель маппера -  скомбинировать новый объект, а этот сервис смешивает маппинг и запрос инфо из бд

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -0,0 +1,23 @@
+import java.util.Collection;
+import java.util.List;
+
+public interface FilmService {
```
Лучше разделить интерфейс на 2: для крудов и для лайков

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/service/UserService.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.Collection;
+import java.util.Optional;
+
+public interface UserService {
```
Лучше разделить интерфейс на 2: для крудов и для друзей

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/DbFilmServiceImpl.java:
```diff
@@ -0,0 +1,61 @@
+
+@Slf4j
+@Service("dbFilmService")
+public class DbFilmServiceImpl implements FilmService {
```
Этот сервис выглядит бесполезным, он просто делегирует все запросы репозиторию

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorageImpl.java:
```diff
@@ -0,0 +1,155 @@
+        Optional<Film> resultFilm = Optional.ofNullable(jdbcTemplate.queryForObject(sqlQuery,
+                filmRowMappers::mapRowToFilm, id));
+
+        if (resultFilm.isPresent()) {
```
Optional лучше всегда использовать с orElse orElseThrow

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmDbStorageImpl.java:
```diff
@@ -0,0 +1,155 @@
+@RequiredArgsConstructor
+public class FilmDbStorageImpl implements FilmStorage {
+
+    private final JdbcTemplate jdbcTemplate;
```
В репозитории не должно быть зависимостей от других репозиториев

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmLikeDbStorageImpl.java:
```diff
@@ -0,0 +1,150 @@
+public class FilmLikeDbStorageImpl implements FilmLikeStorage {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final FilmStorage filmStorage;
```
В репозитории не должно быть зависимостей от других репозиториев

Commit:
12 спринт финал
без правок но с тестами из 11 спринта

Commit:
12 спринт финал
разделение фильм сервиса на круды и лайки

Commit:
12 спринт финал
разделение фильм сервиса на круды и лайки

Commit:
12 спринт финал
разделение фильм сервиса на круды и лайки

Commit:
12 спринт финал
разделение фильм сервиса на круды и лайки

Commit:
12 спринт финал
разделение фильм сервиса на круды и лайки

Commit:
12 спринт финал
замена isPresent на orElseThrow

Commit:
12 спринт финал
Работа с репозиториями и попытка снизить нарушение SRP путем дублирования, далее лучше заменить сервисами отдельными, но чтобы все не напутать, пока так

Commit:
12 спринт финал
Работа с репозиториями и попытка снизить нарушение SRP путем дублирования, далее лучше заменить сервисами отдельными, но чтобы все не напутать, пока так

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/GenreController.java:
```diff
@@ -0,0 +1,30 @@
+@RequiredArgsConstructor
+public class GenreController {
+
+    private final DbGenreServiceImpl service;
```
Все поля лучше по возможности делать интерфейсами

pavelrk97 replied:
Так контроллер это же и так как интерфейс уже? Если возможно, я б уже так оставил, хотелось время найти свой проект попробовать начать)

NikolaiDem replied:
Имелось ввиду GenreService

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/mappers/FilmRowMappers.java:
```diff
@@ -0,0 +1,76 @@
+//@Slf4j
+//@Component
+//@RequiredArgsConstructor
+//public class FilmRowMappers {
```
Удалить комментарии

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/DbFriendshipServiceImpl.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Override
+    public Collection<User> getFriends(Long userId) {
+        if (userCrudService.findById(userId).isEmpty()) {
```
С Optional лучше использовать orElseThrow(...)

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FilmGenreDbStorageImpl.java:
```diff
@@ -0,0 +1,107 @@
+public class FilmGenreDbStorageImpl implements FilmGenreStorage {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final GenreStorage genreStorage;
```
В репозитории не должно быть зависимостей от других репозиториев

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FriendshipDbStorageImpl.java:
```diff
@@ -0,0 +1,139 @@
+public class FriendshipDbStorageImpl implements FriendshipStorage {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final UserStorage userStorage;
```
В репозитории не должно быть зависимостей от других репозиториев

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FriendshipDbStorageImpl.java:
```diff
@@ -0,0 +1,139 @@
+            throw new ValidationException("Нельзя добавить в друзья самого себя");
+        }
+
+        Optional<User> mainUser = userStorage.findById(user1Id);
```
orElseThrow(...)

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/storage/impl/FriendshipDbStorageImpl.java:
```diff
@@ -0,0 +1,139 @@
+                "JOIN friendship AS fri2 ON users.id = fri2.user2_id " +
+                "WHERE fri.user1_id = ? AND fri2.user1_id = ? ";
+
+        if (userStorage.findById(firstUserId).isPresent() && userStorage.findById(secondUserId).isPresent()) {
```
Везде orElseThrow(...)

Commit:
12 спринт финал
Доделаны репозитории

Commit:
12 спринт финал
Доделаны репозитории

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/MpaController.java:
```diff
@@ -0,0 +1,30 @@
+@RequiredArgsConstructor
+public class MpaController {
+
+    private final DbMpaServiceImpl service;
```
И здесь тоже интерфейс

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/DbGenreServiceImpl.java:
```diff
@@ -0,0 +1,38 @@
+
+@Slf4j
+@Service
+public class DbGenreServiceImpl implements GenreService {
```
Реализация сервиса в принципе никак не связана с db
